### PR TITLE
Machine info run III

### PR DESCRIPTION
**Pull request by JoschD**
fixed. merge before release?

### CODE REVIEW

**Comment by JoschD** [comment]
Examples:
```
from pylhc.machine_settings_info import get_info
from omc3.utils import logging_tools

LOG = logging_tools.get_logger(__name__, level_console=logging_tools.DEBUG, color=True)

get_info(
    time="2018-10-30 11:46:00.000",  # Last RunII MD
#     time="2021-10-31 10:46:00.000",  # This Sunday
    knobs=[
        # "LHCBEAM/IP1-XING-H-MURAD",
        "LHCBEAM/IP1-XING-V-MURAD",
        "LHCBEAM/IP2-XING-V-MURAD",
        "LHCBEAM/IP5-XING-H-MURAD",
        # "LHCBEAM/IP5-XING-V-MURAD",
        "LHCBEAM/IP8-XING-H-MURAD",

        "LHCBEAM/IP1-SEP-H-MM",
        # "LHCBEAM/IP1-SEP-V-MM",
        "LHCBEAM/IP2-SEP-H-MM",
        # "LHCBEAM/IP5-SEP-H-MM",
        "LHCBEAM/IP5-SEP-V-MM",
        "LHCBEAM/IP8-SEP-V-MM",

        # "LHCBEAM/IP5-OFFSET-V-MM",
    ],
    output_dir="temp_machine_info_output",
    log=True,
    knob_definitions=True,
)
```

File: pylhc/data_extract/lsa.py [hunk 1]
```diff
@@ -100,42 +140,64 @@ def find_trims_at_time(
...
 
         Returns:
             Dictionary with context info.
         """
-        bp = self._contextService.findStandAloneBeamProcess(beamprocess)
-        return _beamprocess_to_dict(bp)
+        if isinstance(beamprocess, str):
+            beamprocess = self._contextService.findStandAloneBeamProcess(beamprocess)
+        bp_dict = _beamprocess_to_dict(beamprocess)
+        LOG.debug(str(bp_dict))
```
Comment (fsoubelet) [inline pylhc/data_extract/lsa.py:170]: Is this log statement forgotten or wanted?

Comment (JoschD) [reply pylhc/data_extract/lsa.py:170]: wanted :)

File: pylhc/machine_settings_info.py [hunk 2]
```diff
@@ -154,77 +175,89 @@ def log_summary(acc_time: AccDatetime, bp_info: DotDict, o_info: DotDict, trims:
     """Log the summary."""
```
Comment (fsoubelet) [inline pylhc/machine_settings_info.py:175]: I would like to see a docstring here for the different arguments.

Comment (fsoubelet) [reply pylhc/machine_settings_info.py:175]: Also maybe default the arguments to `None` as this is the performed check later on

File: pylhc/machine_settings_info.py [hunk 3]
```diff
@@ -154,77 +175,89 @@ def log_summary(acc_time: AccDatetime, bp_info: DotDict, o_info: DotDict, trims:
...
+
     summary += "-----------------------------------------\n\n"
     LOG.info(summary)
 
 
 def write_summary(
-    output_path: str, acc_time: AccDatetime, bp_info: DotDict, o_info: DotDict, trims: dict
+    output_path: Path, acc_time: AccDatetime, bp_info: DotDict, o_info: DotDict, trims: dict
 ):
     """Write summary into a **tfs** file."""
```
Comment (fsoubelet) [inline pylhc/machine_settings_info.py:203]: Same here

**Review by fsoubelet — APPROVED** [review]

**Pull request merged**